packages/core/platform/platform.js

```diff
@@ -152,29 +150,102 @@ const parseUserAgent = (userAgent) => {
+			// Note that we don't catch 'Tablet' of Firefox as it can't be normalized with other browsers
+			detectedInfo.type = 'mobile';
+		} else {
+			detectedInfo.type = 'desktop';
+		}
+
+		detectedInfo[detectedInfo.browserName] = detectedInfo.browserVersion;
+	}
+
+	// deprecation warning for browser versions older than our support policy
+	if (supportedVersions[detectedInfo.browserName] > detectedInfo.browserVersion) {
+		deprecate({name: `supporting ${detectedInfo.browserName} version before ${supportedVersions[detectedInfo.browserName]}`, until: '5.0.0'});
+	}
+
+	// Merge legacy platform info
+	return {...parseUserAgentLegacy(userAgent), ...detectedInfo};
+};
+
 /**
  * @typedef {Object} PlatformDescription
- * @property {Object} [extra] - Additional information about the detected platform
- * @property {Boolean} gesture - `true` if the platform has native double-finger events
- * @property {Boolean} node - `true` only if `window` is `undefined`
- * @property {String} [platformName] - The name of the platform, if detected
- * @property {Boolean} touch - `true` if the platform has native single-finger events
- * @property {Boolean} touchscreen - `true` if the platform has a touch screen
- * @property {Boolean} unknown - `true` for any unknown system
+ * @property {String} browserName - The name of the detected browser
+ * @property {Number} browserVersion - The version of the detected browser
+ * @property {Number} chrome - The version of the detected browser, if chrome browser is detected
+ * @property {Object} [extra] - Additional information about the detected platform. Deprecated: will be removed in 5.0.0.
+ * @property {Number} firefox - The version of the detected browser, if firefox browser is detected
+ * @property {Boolean} gesture - `true` if the platform has native double-finger events. Deprecated: will be removed in 5.0.0.
+ * @property {Boolean} node - `true` only if `window` is `undefined`. Deprecated: will be removed in 5.0.0. Use `type` instead.
+ * @property {String} [platformName] - The name of the platform, if detected. Deprecated: will be removed in 5.0.0. Use `browserName` instead for browser names.
+ * @property {Number} safari - The version of the detected browser, if safari browser is detected
+ * @property {Boolean} touch - `true` if the platform has native single-finger events. Deprecated: will be removed in 5.0.0. Use `touchEvent` instead.
+ * @property {Boolean} touchEvent - `true` if the browser has native touch events
+ * @property {Boolean} touchscreen - `true` if the platform has a touch screen. Deprecated: will be removed in 5.0.0. Use `touchScreen` instead.
+ * @property {Boolean} touchScreen - `true` if the platform has a touch screen
+ * @property {String} type - The type of the detected platform. One of 'desktop', 'mobile', 'webos', 'node', or 'unknown'
+ * @property {Boolean} unknown - `true` for any unknown system. Deprecated: will be removed in 5.0.0. Use `type` instead.
```

Comment:
I also considered that ordering, but we normally explain props in alphabetical order. 🤔

Reply:
@seunghoh, I think it would make more sense to put deprecated properties at the bottom to let developers know which properties to use. But since we normally place the props in alphabetical order, I'll leave it as is.
CC. @0x64

Reply:
@0x64 Sorry, I changed my mind :( 
Let's move deprecated props to the bottom for enactjs.com API  page

Reply:
@seunghoh @juwonjeong As far as I know, the order of comments is not the same as the order of documents. So, I cannot control it.